### PR TITLE
Add SameIntervalMergeTask for easier usage of MergeTask

### PR DESCRIPTION
**Pull request by kaijianding**
We generate several small segments each hour in realtime node, and want to merge all yesterday"s segments to only 1 segment.
MergeTask needs DataSegment list to be provided, which is painful in actual use.

### CODE REVIEW

**Comment by gianm** [comment]
@kaijianding, this generally looks good to me. Could you include a test, ideally one for behavior and one for serde of the task itself?

**Comment by kaijianding** [comment]
ut added, @gianm

File: indexing-service/src/main/java/io/druid/indexing/common/task/Task.java [hunk 1]
```diff
@@ -58,7 +58,9 @@
     @JsonSubTypes.Type(name = "version_converter", value = ConvertSegmentBackwardsCompatibleTask.class), // Backwards compat - Deprecated
     @JsonSubTypes.Type(name = "version_converter_sub", value = ConvertSegmentBackwardsCompatibleTask.SubTask.class), // backwards compat - Deprecated
     @JsonSubTypes.Type(name = "convert_segment", value = ConvertSegmentTask.class),
-    @JsonSubTypes.Type(name = "convert_segment_sub", value = ConvertSegmentTask.SubTask.class)
+    @JsonSubTypes.Type(name = "convert_segment_sub", value = ConvertSegmentTask.SubTask.class),
+    @JsonSubTypes.Type(name = "same_interval_merge", value = SameIntervalMergeTask.class),
+    @JsonSubTypes.Type(name = "same_interval_merge_sub", value = SameIntervalMergeTask.SubTask.class)
```
Comment (himanshug) [inline indexing-service/src/main/java/io/druid/indexing/common/task/Task.java]: why is SubTask needed here?

Comment (kaijianding) [inline indexing-service/src/main/java/io/druid/indexing/common/task/Task.java]: After test, I verified it is not need here, I removed it from here and also remove other not needed code.

**Comment by gianm** [comment]
still 👍  after latest changes. thanks for identifying and removing the unneeded code.

**Comment by himanshug** [comment]
@kaijianding  you removed overriding of isReady(..) as well ... i thought it was good because segment list is automatically figured out so no need to fetch segment list again and match as is done in MergeTaskBase.isReady(..) ?

**Comment by kaijianding** [comment]
SubTask.run() is called directly inside SameIntervalMergeTask.run() after SameIntervalMergeTask.isReady() called.
In this way, I think SubTask is not submitted to task queue, so the `isReady` is not going to be called, that's why I removed `isReady` from SubTask. @himanshug

**Comment by himanshug** [comment]
@kaijianding yeah, its fine then.

**Pull request merged**